### ISSUE TITLE
HTTP Authentication Exclude HLS Protocol

### ISSUE THREAD

**Issue by Craytor** [bluenviron/mediamtx#3770]
## Which version are you using?
v1.9.0

## Which operating system are you using?
- [ ] Linux amd64 standard
- [x] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue
When using `authHTTPExclude`, I am providing the "protocol" of "hls" of which I would like to prevent authentication due to performance concerns and the overall use case of hls http authentication. However, this does not appear to be preventing hls authentication from occurring as I am still getting requests to my server. This may not be a bug, but I believe this functionality would be great to have and is very similar to the "action" type for what to exclude.

## Describe how to replicate the issue
```
authMethod: http
authHTTPAddress: http://address
authHTTPExclude:
  - protocol: hls
```

1. Start the server with the supplied config for auth
2. Begin publishing with streaming client
3. Begin viewing stream

During this time, the sever will attempt to verify each request by sending it to the http address. However, I followed the example of authHTTPExclude included an action with a type. The request still goes to the HTTP server.

## Did you attach the server logs?
No

## Did you attach a network dump?
No

**Comment by jwetzell** [comment]
I don't believe `protocol` is valid property there. The definition for that has `action` and `path`. It does not look possible to exclude auth for a particular protocol.

**Comment by aler9** [comment]
moved in #4103
